### PR TITLE
Collective sections format retro-compatibility

### PR DESCRIPTION
**Pull request by Betree**
Require https://github.com/opencollective/opencollective-frontend/pull/5575
Requirement for https://github.com/opencollective/opencollective/issues/3773

- Adapted preload
- Made sure navbar still works if you're using a frontend with old navbar while the collective has new sections format

### CODE REVIEW

File: lib/collective-sections.js [hunk 1]
```diff
@@ -1,20 +1,21 @@
 /**
- * There are 3 levels of filtering to know if a section should appear or not.
+ * There are 3 levels of filtering to know if a section should appear or not:
  *
  * 1. Status of the collective: the `type`, `isActive`
  * 2. User's permissions: certain sections are displayed only to admins
  * 3. The data: we won't display a section if it's empty
+ *
+ * The logic for these checks is progressively moving to the API, relying on the "features"
```
Comment (sbinlondon) [inline lib/collective-sections.js:8]: 👍

File: lib/collective-sections.js [hunk 2]
```diff
@@ -445,6 +445,45 @@ export const convertSectionsToNewFormat = sections => {
   return convertedSections;
 };
 
+/**
+ * Converts sections back to legacy format ([{ section, isEnabled }]), without categories
+ */
+export const convertSectionsToLegacyFormat = sections => {
```
Comment (sbinlondon) [inline lib/collective-sections.js:451]: Great - so now we can convert both ways? 🎉

**Review by sbinlondon — APPROVED** [review]
Looks good to me! Just needs to rebase against the latest version of my PR with `navbarVersion` instead of `version` and the changes from `withRouter` to `useRouter` for some of the components.

File: lib/collective-sections.js (hunk 2 above)
Comment (Betree) [reply lib/collective-sections.js:451]: Exactly :-)

Comment (Betree) [reply lib/collective-sections.js:451]: Old frontend will work with new format

**Pull request merged**